warehouse/forklift/legacy.py

```diff
@@ -400,6 +400,54 @@ def _validate_classifiers(form, field):
             )
 
 
+def _validate_dynamic(form, field):
+    dynamic_fields = set(map(str.lower, field.data or []))
+
+    disallowed_dynamic_fields = []
+
+    for field_name in ("Name", "Version", "Metadata-Version"):
+        if field_name.lower() in dynamic_fields:
+            disallowed_dynamic_fields.append(field_name)
+
+    if disallowed_dynamic_fields:
+        if len(disallowed_dynamic_fields) == 1:
+            values = f"field {disallowed_dynamic_fields[0]!r}"
+        else:
+            values = f"fields {disallowed_dynamic_fields!r}"
+
+        raise wtforms.validators.ValidationError(
+            f"The {values} cannot be marked as dynamic.",
+        )
+
+
+_extra_name_re = re.compile("^([a-z0-9]|[a-z0-9]([a-z0-9-](?!--))*[a-z0-9])$")
+
+
+def _validate_extras(form, field):
+
+    # TODO: is metadata_version guaranteed to be valid by this point?
+    metadata_version = packaging.version.Version(form.metadata_version.data)
```

Comment:
No, but an invalid version will be the first message displayed (https://github.com/pypi/warehouse/blob/main/warehouse/forklift/legacy.py#L852-L854) so I think it's fine to just try/catch here and raise a `ValidationError` here if the version does not parse.